cmake/one_core_compliance.cmake

```diff
@@ -0,0 +1,36 @@
+#has to be set before project
+if(MSVC_ONE_CORE_COMPLIANCE)
```

Comment:
can we move content of this script to `cmake/onecoreuap.toolchain.cmake`? In this case we can pass `-DCMAKE_TOOLCHAIN_FILE=cmake/onecoreuap.toolchain.cmake` and don't need `MSVC_ONE_CORE_COMPLIANCE` variable, don't need to set with force `CMAKE_C_STANDARD_LIBRARIES` and `CMAKE_CXX_STANDARD_LIBRARIES`

Reply:
very likely - yes. I need to check side effects

Reply:
I have created toolchain based on these changes https://github.com/openvinotoolkit/openvino/pull/3044. @pshemoth could you please try and give feedback?